integration-tests/minio/src/test/java/org/apache/camel/quarkus/component/minio/it/MinioTest.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.minio.it;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.junit.QuarkusTest;
+import io.restassured.RestAssured;
+import io.restassured.http.ContentType;
+import io.restassured.response.ValidatableResponse;
+import io.restassured.specification.RequestSpecification;
+import org.apache.camel.component.minio.MinioConstants;
+import org.apache.camel.component.minio.MinioOperations;
+import org.apache.camel.util.CollectionHelper;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+
+@QuarkusTest
+@QuarkusTestResource(MinioTestResource.class)
+class MinioTest {
+
+    private final String BUCKET_NAME = "mycamel";
+
+    @Test
+    public void testConsumer() throws Exception {
+        initClient();
+
+        ExecutorService executor = Executors.newSingleThreadExecutor();
+
+        Future<String> fr1 = executor.submit(
+                () -> RestAssured.get("/minio/consumer")
+                        .then()
+                        .extract().asString());
```

Comment:
Do we really need the Executor? `consumerTemplate.receiveBody(String endpointUri, long timeout, Class<T> type)` is blocking so I guess you can put `RestAssured.get("/minio/consumer")` directly to the `untilAsserted()` block below.

Reply:
thanks for the advice, I'll try it that way.

Reply:
done